src/conmon.c

```diff
@@ -1750,13 +1750,25 @@ int main(int argc, char *argv[])
 		closedir(fdsdir);
 	}
 
+	_cleanup_free_ char *status_str = g_strdup_printf("%d", exit_status);
+
+	/* Always write the exit file to container directory */
+	_cleanup_free_ char *ctr_exit_file_path = g_build_filename(opt_bundle_path, "exit", NULL);
+	if (!g_file_set_contents(ctr_exit_file_path, status_str, -1, &err))
+		nexitf("Failed to write %s to container exit file: %s", status_str, err->message);
+
+	/*
+	 * Writing to this directory helps if a daemon process wants to monitor all container exits
+	 * using inotify.
+	 */
 	if (opt_exit_dir) {
-		_cleanup_free_ char *status_str = g_strdup_printf("%d", exit_status);
 		_cleanup_free_ char *exit_file_path = g_build_filename(opt_exit_dir, opt_cid, NULL);
-		if (!g_file_set_contents(exit_file_path, status_str, -1, &err))
-			nexitf("Failed to write %s to exit file: %s", status_str, err->message);
+		if (link(ctr_exit_file_path, exit_file_path) == -1) {
```

Comment:
this works only if both paths are on the same file system.  Can we be sure about it?

It would be safer to fallback to `g_file_set_contents` if the `link(2)` fails

Reply:
Good point @giuseppe, I will add the fallback on failure.